helper.js

```diff
@@ -1,25 +1,42 @@
 const fs = require("fs-extra");
 const path = require("path");
 
-const dataDirPath = path.resolve(__dirname, "./temp/data");
-const attachDirPath = path.resolve(__dirname, "./temp/images");
+let dataDirPath = '';
+let attachDirPath = '';
+let tempDirPath = path.resolve(__dirname, "./temp");
+
+async function setUpPaths(pathToDirectory) {
+  if (pathToDirectory !== undefined) {
+    tempDirPath = path.resolve(pathToDirectory, "./temp");
+  }
+
+  dataDirPath = path.resolve(tempDirPath, "./data");
+  attachDirPath = path.resolve(tempDirPath, "./images");
+
+  await fs.remove(tempDirPath);
+  await fs.mkdirs(dataDirPath);
+  await fs.mkdirs(attachDirPath);
+};
 
 const distDirName = `./jest-html-reporters-attach`;
 
 /**
  *
  * @param {Buffer | string} attach
  * @param {string} description
+ * @param {object} context. Optional. It contains custom configs
  */
 const addAttach = async (attach, description, context) => {
-  const { testPath, testName } = getJestGlobalData(context);
+  await setUpPaths(context.tempDirPath);
+  const { testPath, testName } = getJestGlobalData(context.context);
```

Comment:
The param `context` is used here  https://github.com/Hazyzh/jest-html-reporters/pull/84, I am afraid the change here will break the original feature.

Reply:
Well, on the one hand, you are right.
On the other hand, I'm not a big fan of making methods with 3+ arguments, so I think it's acceptable to reimplement it now and pass all optional params inside this object. It provides an opportunity to extend it in the future if it will be needed.

Reply:
sure, you can change the implementation make the param as an object. 

Also, for this specific scenario, how about using the environment variable? 
something like this 
`const dataDirPath = path.resolve(process.env[JEST_HTML_REPORTERS_TEMP_DIR_PATH] || __dirname, "./data");`

Reply:
yeah, I think for this scenario it will be fine.
in this case, I'll add `JEST_HTML_REPORTERS_TEMP_DIR_PATH ` option to config and make `context` param as object